### ISSUE TITLE
Consider using `ManagedResource`s for deploying shoot control plane components instead of directly applying the resources with the client

### ISSUE THREAD

**Issue by rfranzke** [gardener/gardener#8241]
/area dev-productivity
/kind enhancement

For historic reasons, shoot control plane components are applied directly via the client ([example](https://github.com/gardener/gardener/blob/c6d41732d58e221abba2db5ec67dfe75ff8c72d4/pkg/component/kubescheduler/kube_scheduler.go#L211-L231)) while system components are deployed via `ManagedResource`s ([example](https://github.com/gardener/gardener/blob/c6d41732d58e221abba2db5ec67dfe75ff8c72d4/pkg/component/coredns/coredns.go#L131-L138)).

Using `ManagedResource`s has multiple benefits, most prominently
- automatic clean-up of no longer needed resources
- health checks
- progressing checks
- convenient application (no need to check for unintended field overwrites (e.g., when a resource is scaled via HPA))

Hence, we should think about transitioning also to `ManagedResource`s for the shoot control plane components.

Before we start, we should evaluate:
- Are there any downsides/drawbacks when we do so (e.g., with respect to extension webhooks that try to modify things)?
- Should we switch to a custom resource instead of using `Secret`s for the resources part of `ManagedResource`s?

**Comment by dimityrmirchev** [comment]
If you discuss switching to a custom resource instead of `Secret`s please consider the problems tackled in https://github.com/gardener/gardener/pull/7959 and https://github.com/gardener/gardener/pull/8116, i.e. that right now there is no reliable mechanism to check if the latest version of the resources is applied or not.

**Comment by rfranzke** [comment]
Update: We probably still want this, yet we're currently focusing on other major topics (i.e., we don't have the capacity to work on this right now). If somebody would like to volunteer, please get in touch.

**Comment by aniruddha2000** [comment]
/assign 

would like to work on it
